src/main/jbake/content/blog/2016/12/19/hawkular-metrics-compression.adoc

```diff
@@ -0,0 +1,93 @@
+= Hawkular-Metrics and the new compression algorithm
+Michael Burman
+2016-12-19
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, metrics, compression, gorilla, storage, performance
+
+In Hawkular-Metrics release 0.20.0 we introduced a new compression capability that reduces the required storage space. This blog post will describe details about the implementation in the processing pipelines and what implications there are for the user. Should you want to skip the technical details, head over to the <<benchmark,benchmark section>> below.
+
+Some of the details mentioned here were improved or introduced in the 0.21.0, but the compression ratio is equal between 0.20.0 and 0.21.0. Results are published against 0.22.0 which is the newest released version. 
+
+=== Compression algorithm
+
+In this release, we support the compression from the Facebook's great link:www.vldb.org/pvldb/vol8/p1816-teller.pdf[Gorilla paper] with small changes. For complete overview on how the compression method works I recommend reading the original paper. For short overview, the timestamps are compressed by calculating the delta of delta and writing them with variable length encoding while the data values are compressed by taking a XOR (eXclusive OR) between values (using the previous value as predictor for the next one) and storing that with variable length encoding also. We also slightly change the implementation details, as we need to use millisecond precision instead of the second precision used by Facebook.
```

Comment:
"For complete.." -> For a complete (?)

"delta of delta" sounds strange. Perhaps quickly explain what it means (provide an example?)

Perhaps first mention that an item consists of ts, value, ??